connector_elasticsearch/components/adapter.py

```diff
@@ -79,10 +86,18 @@ def delete(self, binding_ids):
                 "_id": binding_id,
             }
             dataforbulk.append(action)
-
-        res = elasticsearch.helpers.bulk(es, dataforbulk)
+        try:
+            res = elasticsearch.helpers.bulk(es, dataforbulk)
+        except elasticsearch.helpers.errors.BulkIndexError as e:
+            # if the document we are trying to delete does not exist,
+            # we can consider deletion a success (there is nothing to do).
+            if not _is_delete_nonexistent_documents(e):
+                raise e
+            msg = "Trying to delete non-existent documents. Ignored: %s"
+            _logger.warning(msg, e)
+            skip_check = True
         # checks if number of indexed object and object in datas are equal
-        return len(binding_ids) - res[0] == 0
+        return skip_check or len(binding_ids) - res[0] == 0
```

Comment:
nit instead of skipping the check, we could count the delete/404 in res. Or remove the check completely ? Are there callers who care about the result of this method ?